File: backend/src/main/java/dk/mathiaskofod/services/session/actions/lobby/participant/SendEmojiAction.java
package dk.mathiaskofod.services.session.actions.lobby.participant;

import dk.mathiaskofod.services.lobby.models.Emoji;
import dk.mathiaskofod.services.session.models.annotations.ActionType;

@ActionType("SEND_EMOJI")
public record SendEmojiAction(Emoji emoji) implements LobbyParticipantAction {}

File: frontend/src/app/services/models/categories/actions/lobby/common/send-emoji-action.ts
import {LobbyAction} from '../lobby-action';
import {Emoji} from '../../../../../../../api-models/model/emoji';

interface SendEmojiAction extends LobbyAction {
  emoji: string;
}

export function sendEmojiAction(emoji: Emoji): SendEmojiAction {
  return {emoji: emoji, type: "SEND_EMOJI"}
}
